### ISSUE TITLE
Web FastPass Tab stops view and report fit and finish inconsistencies

### ISSUE THREAD

**Issue by dbjorge** [microsoft/accessibility-insights-web#5099]
## Describe the bug

This issue tracks a collection of small fit and finish issues discovered during release validation for `feat(tabstops-report)` in Insider 2.30.0.

#### Summary bar

* [x] (#5109 The Summary bar is intended to be contained inside a white "card" rectangle
* [x] (#5109) The "x" and "check" icons are misaligned; they should be vertically aligned with the numbers next to them
* Note that the icons' colors are mismatched from the Figma (the check and x should be colored icons in a white circle, not the current white icons in a colored circle). However, they are mismatched the same way in the Assessment report; for consistency, they can stay as-is.

#### Heading levels
 
* [x] (#5121) The heading level for "Failed Instances"/"Passed checks"/"Incomplete checks" is h2, should be h3
* [x] (#5121) The heading level for failed instance cards is h3, should be h4
* Note: Be sure not to accidentally break heading levels in other reports, may need to parameterize levels
* Note: Heading weights and sizes can stay as-is, despite mismatching Figma slightly, since the current versions meet the Heading Level assessment requirement and the Figma version does not

#### Spacing between sections
 
* [x] (#5119) Spacing between comments in tab stops failure instance should be bigger; it should match the details view UI
* [x] (#5119) Spacing between "failed instances", "incomplete checks", "passed checks" sections should be smaller (see figma)
* [x] (#5119) Spacing between "Automated checks" and "Tab stops" sections should be bigger, see Figma (be careful to keep the "Automated checks haven't been run" case consistent)

### Details view FastPass Tab stops UI

* [x] (#5107) Changing a pass/fail option control from "unselected" to "selected" currently causes the containing row to change height slightly, such that all components underneath it shifts downward slightly. This shouldn't happen.
* [x] (#5111) Rows for individual comments should have an on-hover gray background effect similar to the rows in instance tables in assisted Assessment requirements
* [x] (#5110) The report and the details view should be consistent about whether there is a colon after the word "Comment"
* [x] (#5115) At window widths from 600px to 704px, a spurious horizontal scroll bar appears for the `DetailsList` Requirements table. This is not technically a WCAG violation since this qualifies as a data table, but it is still an unnecessary usability issue.
* [x] (#5115) At window widths from 600px to 678px, the above issue also occurs with the `DetailsList` for failed instances' comments.
* [x] (#5115) Comment edit/delete icons should be aligned with requirement pass/fail options

## To Reproduce

1. Install Insider 2.30.0
2. Start a FastPass on any test page with issues
3. Navigate to the Tab stops requirement
4. Add a mix of pass/fail results, including at least one case with multiple comments
5. Observe issues from "Details view FastPass Tab stops UI" section above
6. Export a report
7. Observe issues from "FastPass report" section above

## Screenshots

*Screenshot with checks next to correct headings and annotations for incorrect headings in current report*:
![screenshot of heading level issues](https://user-images.githubusercontent.com/376284/150616833-a4ecc18a-bfb5-4982-8d5a-3ee1b7794036.png)

*Screenshot of spurious horizontal scroll bars*: 
![Screenshot highlighting the extra scrollbars described above](https://user-images.githubusercontent.com/376284/150617267-87a51647-f484-46f7-adaa-34cf7d2dc06c.png)

## Context (please complete the following information)

- OS Name & Version: Windows 11 22000.434
- AI-Web Version & Environment: 2.30.0 Insider
- Browser Version: Edge stable 97.0.1072.69 (Official build) (64-bit)
- Target Page: n/a

## Are you willing to submit a PR?

Yes

## Did you search for similar existing issues?

Yes

## Additional context

n/a

**Comment by ferBonnin** [comment]
@dbjorge, @jalkire is this bug fixed?

**Comment by dbjorge** [comment]
Yes
